turbopack/crates/turbopack-core/src/chunk/chunk_group.rs

```diff
@@ -141,16 +142,20 @@ pub async fn make_chunk_group(
         .map(|&loader| loader.references())
         .try_join()
         .await?;
-    let async_loader_external_module_references = async_loader_references
-        .iter()
-        .flat_map(|references| references.iter().copied())
-        .map(|v| *v)
-        .collect();
+    let async_loader_external_module_references = Vc::cell(
+        async_loader_references
+            .iter()
+            .flat_map(|references| references.iter().copied())
+            .collect(),
+    );
 
-    let mut referenced_output_assets = references_to_output_assets(external_module_references)
-        .await?
-        .await?
-        .clone_value();
+    let mut referenced_output_assets = (*external_output_assets.await?).clone();
+    referenced_output_assets.extend(
+        references_to_output_assets(external_module_references)
+            .await?
+            .await?
+            .clone_value(),
+    );
```

Comment:
Using an iterator might be slightly faster because it can't reuse the cloned vector anyway
```suggestion
    referenced_output_assets.extend(
        references_to_output_assets(external_module_references)
            .await?
            .await?
            .iter()
            .copied()
    );
```